pkg/errutil/multierror.go

```diff
@@ -32,6 +33,31 @@ func (es MultiError) Err() error {
 	return NonNilMultiError(es)
 }
 
+// SyncMultiError is a thread-safe implementation of MultiError.
+type SyncMultiError struct {
```

Comment:
I think this can be upstreamed to https://github.com/efficientgo/tools/blob/main/core/pkg/merrors/errors.go later on! 🙂